crest/Assets/Crest/Crest/Shaders/OceanInputs/GerstnerShared.hlsl

```diff
@@ -91,3 +88,24 @@ half4 ComputeGerstner(float2 worldPosXZ, float3 uv_slice)
 
 	return _Weight * half4(result, sss);
 }
+
+half4 ComputeShorelineGerstner(float2 worldPosXZ, float3 uv_slice, half depth, half distanceToShore, half2 directionToShore)
+{
+	float2 displacementNormalized = 0.0;
+	half3 result = (half3)0.0;
+
+	if(depth > 0.0)
+	{
+		const float pi = 3.141;
+		const float twoPiOverWavelength = (2.0 * pi) / 2.5;
+		const float twoPiOverPeriod = (2.0 * pi) / 2.0;
+		const float amplitude = 0.12;
+		// Chop increases as depth increases
+		const float chopAmplitude = 2.0/(1.0+sqrt(distanceToShore));
+
+		const float angle = (twoPiOverWavelength * sqrt(distanceToShore)) + (_CrestTime * twoPiOverPeriod);
+		result.y = amplitude * cos(angle);
+		result.xz = chopAmplitude * directionToShore * sin(angle) * result.y;
```

Comment:
Multipliying the chop by result.y means that the waves tip-forward near the top - and it also helps prevent self-intersection (I don't know why! but when I added the term they went away).